cmd/skaffold/app/cmd/util.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/tag"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+)
+
+func getBuildArtifactsAndSetTags(r runner.Runner, config *latest.SkaffoldConfig) ([]build.Artifact, error) {
+	buildArtifacts, err := getArtifacts(fromBuildOutputFile.BuildArtifacts(), preBuiltImages.Artifacts(), config.Build.Artifacts)
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range buildArtifacts {
+		tag, err := r.ApplyDefaultRepo(buildArtifacts[i].Tag)
+		if err != nil {
+			return nil, err
+		}
+		buildArtifacts[i].Tag = tag
+	}
+
+	return buildArtifacts, nil
+}
+
+func getArtifacts(fromFile, fromCLI []build.Artifact, artifacts []*latest.Artifact) ([]build.Artifact, error) {
+	var buildArtifacts []build.Artifact
+	for _, artifact := range artifacts {
+		buildArtifacts = append(buildArtifacts, build.Artifact{
+			ImageName: artifact.ImageName,
+		})
+	}
+
+	// Tags provided by file take precedence over those provided on the command line
+	buildArtifacts = build.MergeWithPreviousBuilds(fromCLI, buildArtifacts)
+	buildArtifacts = build.MergeWithPreviousBuilds(fromFile, buildArtifacts)
+
+	buildArtifacts, err := applyCustomTag(buildArtifacts)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check that every image has a non empty tag
+	_, err = TestValidateArtifacts(buildArtifacts)
```

Comment:
if we're just going to ignore the bool returned here, we probably don't need to return it in the first place. then you can rewrite this as 

```golang
if err := ValidateArtifactTags(buildArtifacts); err != nil {
  return nil, err
}
```

Reply:
Updated in the follow-up pr - https://github.com/GoogleContainerTools/skaffold/pull/5118